### PR TITLE
Amend employment responsibilities

### PR DESCRIPTION
**Pull request by rgforsyth**
## What does this pull request do?

Adds the option for probation practitioners to amend the employment responsibilities.

## What is the intent behind these changes?

Allows referrals to be amended after being sent.

### CODE REVIEW

File: server/routes/amendAReferral/amendAReferralController.test.ts [hunk 1]
```diff
@@ -252,3 +252,72 @@ describe('POST /probation-practitioner/referrals/:id/update-complexity-level', (
...
+  it('redirects to same page with no changes banner when no changes are made', () => {
+    return request(app)
+      .post(`/probation-practitioner/referrals/${referral.id}/employment-responsibilities`)
+      .send({
+        'reason-for-change': 'new value',
+        'has-additional-responsibilities': `${referral.referral.hasAdditionalResponsibilities ? 'yes' : 'no'}`,
+        'when-unavailable': `${referral.referral.whenUnavailable}`,
+      })
+      .expect(302)
+      .expect('Location', `/probation-practitioner/referrals/${referral.id}/employment-responsibilities?noChanges=true`)
```
Comment (kth13) [inline server/routes/amendAReferral/amendAReferralController.test.ts:301]: Here are we not asserting the banner text present?

File: server/routes/amendAReferral/employment-responsibilities/amendEmploymentResponsibilitiesPresenter.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  readonly text = {
+    responsibilities: {
+      title: `Do you want to change whether ${this.sentReferral.referral.serviceUser?.firstName} has caring or employment responsibilities?`,
+      hasAdditionalResponsibilities: {
+        label: `Do you want to change whether ${this.sentReferral.referral.serviceUser?.firstName} has caring or employment responsibilities?`,
+        hint: 'For example, times and dates when they are at work.',
+        errorMessage: this.errorMessageForField('has-additional-responsibilities'),
+      },
+      whenUnavailable: {
+        label: `Provide details of when ${this.sentReferral.referral.serviceUser?.firstName} will not be able to attend sessions`,
```
Comment (kth13) [inline server/routes/amendAReferral/employment-responsibilities/amendEmploymentResponsibilitiesPresenter.ts:36]: In the other amend pages, we are using serviceUser.firstName which I heard the more reliable way to grab the username

File: server/routes/amendAReferral/employment-responsibilities/amendEmploymentResponsibilitiesView.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+  private readonly errorSummaryArgs = ViewUtils.govukErrorSummaryArgs(this.presenter.errorSummary)
+
+  private get textAreaArgs(): TextareaArgs {
+    return {
+      name: AmendEmploymentResponsibilitiesForm.amendEmploymentResponsibilitiesReasonForChangeId,
+      id: AmendEmploymentResponsibilitiesForm.amendEmploymentResponsibilitiesReasonForChangeId,
+      label: {
+        text: this.presenter.text.reasonForChange.title,
+        classes: 'govuk-label--l',
```
Comment (kth13) [inline server/routes/amendAReferral/employment-responsibilities/amendEmploymentResponsibilitiesView.ts:17]: Helen and Emma suggest it to be `govuk-label--m`

File: server/routes/amendAReferral/employment-responsibilities/amendEmploymentResponsibilitiesPresenter.ts (hunk 2 above)
Comment (rgforsyth) [reply server/routes/amendAReferral/employment-responsibilities/amendEmploymentResponsibilitiesPresenter.ts:36]: is it worth the extra call through community api for something we already have access to this way?

File: server/routes/amendAReferral/amendAReferralController.test.ts (hunk 1 above)
Comment (rgforsyth) [reply server/routes/amendAReferral/amendAReferralController.test.ts:301]: The banner text isn't present at the point of the 302. The res text is the location of where the redirect is? Unless there's a way to get it.

res.text gives: "Found. Redirecting to /probation-practitioner/referrals/1/employment-responsibilities?noChanges=true"

**Review by kth13 — APPROVED** [review]
LGTM

**Pull request merged**